pkg/aws/cloud.go

```diff
@@ -25,6 +26,7 @@ type CloudConfig struct {
 	AccountId   string
 	Region      string
 	ClusterName string
+	PrivateVPC  bool
```

Comment:
I think it might be better to be explicit here, as over time this field could be interpreted in different ways. For example, at some stage the Tagging API will likely be enabled for PrivateLink, but then the customer still might not have provisionined the VPC endpoint for Tagging. Once they eventually do, their VPC is still "private" but then they would remove the "PrivateVPC" flag to get to the tagging service, so that semantic doesn't quite feel right.

Instead, we could have ```TaggingServiceAPIDisabled``` which defaults to ```false``` but then for configurations that can't access that service they can set it to ```true```. Later, when they can use the API, they set it back to ```false```.